src/contracts/BCoWHelper.sol

```diff
@@ -61,6 +63,26 @@ contract BCoWHelper is ICOWAMMPoolHelper {
 
     order_ = GetTradeableOrder.getTradeableOrder(params);
 
+    {
+      // NOTE: Using calcOutGivenIn for the sell amount in order to avoid possible rounding
+      // issues that may cause invalid orders. This prevents CoW Protocol back-end from generating
+      // orders that may be ignored due to rounding-induced reverts.
+
+      uint256 balanceToken0 = IERC20(tokens_[0]).balanceOf(pool);
+      uint256 balanceToken1 = IERC20(tokens_[1]).balanceOf(pool);
+      (uint256 balanceIn, uint256 balanceOut) =
+        address(order_.buyToken) == tokens_[0] ? (balanceToken0, balanceToken1) : (balanceToken1, balanceToken0);
+
+      order_.sellAmount = calcOutGivenIn({
+        tokenBalanceIn: balanceIn,
+        tokenWeightIn: 1e18,
+        tokenBalanceOut: balanceOut,
+        tokenWeightOut: 1e18,
```

Comment:
All good, just commenting to make sure I understand why the hardcoded weights don't cause issues.
In our simple case with equal weights, this function only cares about `weightRatio`, and we are sure it's always 1 otherwise `tokens()` would have reverted at a previous step. We could use `IBCoWPool(pool).getNormalizedWeight(token)` here but it's extra conditional logic for no benefits, considering that this will have to be revisited anyway once we start supporting different weights in the same pool.